streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -77,7 +77,9 @@ public void process(K key, Change<V> change) {
             V newValue = computeValue(key, change.newValue);
             V oldValue = sendOldValues ? computeValue(key, change.oldValue) : null;
 
-            if (sendOldValues && oldValue == null && newValue == null) return; // unnecessary to forward here.
+            if ((sendOldValues && oldValue == null && newValue == null) ||
+                (change.oldValue == null && change.newValue == null)) return; // unnecessary to forward here.
```

Comment:
I confirm that you are absolutely right about your last mail about trying to remove sendOldValues from test is an error and I had misunderstood the implication. I also appreciate the debug-friendly syntax of return on next line, I'll adjust that. I'll also remove the change test as per your comment above that spells out you want to handle that in mapValues function. Hopefully, things will be sorted out within an hour or so.